### ISSUE TITLE
tmpfs crash upon reboot

### ISSUE THREAD

**Issue by martelletto** [bitrig/bitrig#6]
syncing disks... panic: kernel diagnostic assertion "mp->tm_bytes_used == 0" failed: file "../../../../tmpfs/tmpfs_mem.c", line 66
Stopped at      Debugger+0x5:   popq    %rbp
RUN AT LEAST 'trace' AND 'ps' AND INCLUDE OUTPUT WHEN REPORTING THIS PANIC!
IF RUNNING SMP, USE 'mach ddbcpu <#>' AND 'trace' ON OTHER PROCESSORS, TOO.
DO NOT EVEN BOTHER REPORTING THIS WITHOUT INCLUDING THAT INFORMATION!
ddb{0}> trace
Debugger() at Debugger+0x5
panic() at panic+0x131
__assert() at __assert+0x24
tmpfs_mntmem_destroy() at tmpfs_mntmem_destroy+0x2c
tmpfs_unmount() at tmpfs_unmount+0xb3
dounmount() at dounmount+0x83
vfs_unmountall() at vfs_unmountall+0x9e
vfs_shutdown() at vfs_shutdown+0x41
boot() at boot+0x73
sys_reboot() at sys_reboot+0x59
syscall() at syscall+0x225
--- syscall (number 55) ---
end of kernel
end trace frame: 0x7f7ffffc4470, count: -11
        acpi_pdirpa+0x3fc7fa:
acpi_pdirpa+0x3fc7fa:
ddb{0}> ps
   PID   PPID   PGRP    UID  S       FLAGS  WAIT          COMMAND
*31142      1  31142  31539  7        0x22                reboot
    21      0      0      0  3    0x100200  aiodoned      aiodoned
    20      0      0      0  2    0x100200                update
    19      0      0      0  3    0x100200  cleaner       cleaner
    18      0      0      0  3    0x100200  reaper        reaper
    17      0      0      0  3    0x100200  pgdaemon      pagedaemon

**Comment by Bluerise** [comment]
Is this still relevant?

**Comment by martelletto** [comment]
Yes, it is. I had the issue diagnosed a long time ago, but forgot to write it down. Unfortunately it is very hard to reproduce.

**Comment by 4mig4** [comment]
Is this relevant now ?

**Comment by Bluerise** [comment]
Apparently so, even though it only happens rarely?
